website/src/lib/oasst_api_client.ts

```diff
@@ -351,6 +351,23 @@ export class OasstApiClient {
     return this.get<Message[]>(`/api/v1/messages?${params}`);
   }
 
+  fetch_my_messages_cursor(
+    user: BackendUserCore,
+    {
+      direction,
+      cursor,
+      ...rest
+    }: { include_deleted?: boolean; max_count?: number; cursor?: string; direction: "forward" | "back"; desc?: boolean }
+  ) {
+    return this.get<FetchUserMessagesCursorResponse>(`/api/v1/messages/cursor`, {
+      ...rest,
+      username: user.id,
+      auth_method: user.auth_method,
+      after: direction === "forward" ? cursor : undefined,
+      before: direction === "back" ? cursor : undefined,
+    });
+  }
+
```

Comment:
There is `fetch_user_messages_cursor`, you dont need to implement this

Reply:
@notmd   Could you help me understand please:

My understanding is have we have both:  `fetch_my_messages` and `fetch_user_messages` because the first one takes a username and auth_method and gets your own messages, authenticating with your own user, and the second takes the user_id and no auth_method only.

So I was following this pattern by adding a `fetch_my_messages_cursor` that also takes a username and auth_method.

I did actually try using `fetch_user_messages_cursor` but it returned a 500 server error, which I assumed was because I was trying to call it as non-admin.

Reply:
ah sorry, those are different endpoints.